### DISCUSSION TITLE
Android Manifest: too much permissions?

### DISCUSSION THREAD

**Discussion by olierxleben** [ionic-team/capacitor#3499] [General] [answered]
Hi all, 

I would like to ask for the intention of the many permissions in the Android template file? I recently got asked for why my App needs all that. I was not aware of the microphone permission? Is it really needed by default capacitor stuff? Cant we minimize "dependencies" here?

**Comment by jcesarmobile** [comment, marked as answer]
If you read the comments on the file, there is a brief description of why are they needed. By default the template includes all the permissions required for al core plugins and some web APIs, but you can remove everything you don’t use/need.
In capacitor 3 we are moving the plugins out, so we will also update the template to not add the permissions.

```
<!-- Permissions -->

    <uses-permission android:name="android.permission.INTERNET" />
    <!-- Camera, Photos, input file -->
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <!-- Geolocation API -->
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-feature android:name="android.hardware.location.gps" />
    <!-- Network API -->
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <!-- Navigator.getUserMedia -->
    <!-- Video -->
    <uses-permission android:name="android.permission.CAMERA" />
    <!-- Audio -->
    <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS"/>
```
